staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -330,17 +330,33 @@ Loop:
 	return p.parseInsideAction(cur)
 }
 
+// findFilterSuffixPosition find the suffix position of filter inside array selection
+func (p *Parser) findFilterSuffixPosition() (int, error) {
+	for i := len(p.input); i > p.pos; i-- {
+		if strings.HasSuffix(p.input[p.pos:i], ")]") {
```

Comment:
There can be more than one filter in a jsonpath. This looks like it only finds the suffix of the last filter.